frontend/src/pages/ShowProfile.tsx

```diff
@@ -0,0 +1,97 @@
+import { Descriptions, DescriptionsProps } from 'antd';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+interface ProfileData {
+  gender: string;
+  occupation: string;
+  favoriteBook: string;
+  description: string;
+  username: string;
+}
+
+function ShowProfile() {
+  const [profileData, setProfileData] = useState<ProfileData | null>(null);
+
+  // Use the useParams hook to get the username from the route
+  const { username } = useParams();
+
+  useEffect(() => {
+    // Fetch profile data when the component mounts
+    const fetchProfileData = async () => {
+      try {
+        //   const response = await fetch(`http://localhost:3001/getProfileData/${username}`);
+        const response = await fetch(
+          `http://localhost:3001/api/getProfileData/${username}`
+        );
```

Comment:
This works fine locally, but will cause issues when we deploy to production

```suggestion
        const response = await fetchProfile(username);
```